### PR TITLE
Add some TIMM micro-benchmarks and some minor benchmark fixes

### PR DESCRIPTION
**Pull request by csarofeen**
Add some benchmarks from TIMM's vision transformer.
Fix benchmarks/.../utils.h path in headers.
Fix printing schedule parameters in benchmark results when fusion segmeneter runs but only makes one kernel.

### CODE REVIEW

File: benchmarks/cpp/nvfuser/timm.cpp [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+#include <torch/csrc/jit/codegen/cuda/scheduler/all_schedulers.h>
+
+#include <benchmark/benchmark.h>
+
+#include <benchmarks/cpp/nvfuser/utils.h>
+
+using namespace torch::jit::fuser::cuda;
+
+static void setup_vit_base_patch16_224_kernel17(Fusion* fusion, void* null) {
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/timm.cpp:14]: Is the `void* null` argument needed?

File: benchmarks/cpp/nvfuser/timm.cpp [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+  std::vector<c10::IValue> aten_inputs({t2, t5, t3, t0, t1});
+  runBenchmarkIterations(benchmark_state, fusion_executor_cache, aten_inputs);
+
+  // inputs: full float, full half, inner dim 3xfloat
+  // outputs: full float, full bool, outer 2 dims 2xfloat, inner dim 2xfloat,
+  // full half Total: full (2xfloat, 2x half, bool), inner dim 5xfloat, outer
+  // dim 2xfloat
+
+  benchmark_state.SetBytesProcessed(
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/timm.cpp:192]: Nit: Replace static data type sizes with `(int64_t) dataTypeSize(at::ScalarType::Float)` for clarity

File: benchmarks/cpp/nvfuser/timm.cpp [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+  auto t2 = at::randn(input_shape, fp32_options);
+  auto t5 = at::randn({input_shape[2]}, fp32_options);
+  auto t3 = at::randn(input_shape, fp16_options);
+  auto t0 = at::randn({input_shape[2]}, fp32_options);
+  auto t1 = at::randn({input_shape[2]}, fp32_options);
+
+  std::vector<c10::IValue> aten_inputs({t2, t5, t3, t0, t1});
+  runBenchmarkIterations(benchmark_state, fusion_executor_cache, aten_inputs);
+
+  // inputs: full float, full half, inner dim 3xfloat
```
Comment (rdspring1) [inline benchmarks/cpp/nvfuser/timm.cpp]: These comments would be useful for the bandwidth calculations in the other tests.

**Review by rdspring1 — APPROVED** [review]
LGTM.

Should we automatically calculate `benchmark_state.SetBytesProcessed` given the fusion inputs and outputs?

File: benchmarks/cpp/nvfuser/timm.cpp (hunk 1 above)
Comment (csarofeen) [reply benchmarks/cpp/nvfuser/timm.cpp:14]: Unfortunately it seems so, the macro's seem to expect at least one argument.

File: benchmarks/cpp/nvfuser/timm.cpp (hunk 2 above)
Comment (csarofeen) [reply benchmarks/cpp/nvfuser/timm.cpp:192]: I think with the comments it should be clear now. Didn't want to add too much verbosity in the size calculation but should be easily translated now.

File: benchmarks/cpp/nvfuser/utils.cpp [hunk 4]
```diff
@@ -142,7 +142,11 @@ void runBenchmarkIterations(
     std::vector<c10::IValue>& aten_inputs) {
   fusion_executor_cache->runFusionWithInputs(aten_inputs);
   bool segmented =
```
Comment (csarofeen) [inline benchmarks/cpp/nvfuser/utils.cpp:144]: @shmsong I think you commented on something similar to this, thought I'd point this out too.

**Pull request merged**